### PR TITLE
LibWeb: Visit the MessagePort's associated Worker target

### PR DESCRIPTION
**Pull request by trflynn89**
Without this, a worker can be GC'd in a very simple script such as:

```js
const worker = new Worker("script.js");
worker.onmessage = () => {};
```

Where script.js attempts to post a message back to the parent window.

When the Worker is GC'd, the IPC connection from the WebContent process to the WebWorker process is closed. When this occurs, the WebWorker will exit() from LibIPC, and any message from the worker to its parent does not have a chance to run.

### CODE REVIEW

**Comment by trflynn89** [comment]
This is a draft for now because I mostly want to see how this fares in CI. This `visit` was changed to an `ignore` due to CI hangs, so let's run it a few times to see if that still occurs.

**Comment by ADKaster** [comment]
Certainly looks like a timeout is still happening on process exit. Not sure if that's a timeout of the WebContent process, or of the headless browser. But clearly something is keeping the headless browser from exiting cleanly after finishing the tests.

**Comment by ADKaster** [comment]
Fixes https://github.com/SerenityOS/serenity/issues/23899

**Review by ADKaster — APPROVED** [review]

File: Userland/Libraries/LibWeb/DOM/Document.cpp [hunk 1]
```diff
@@ -3308,22 +3309,35 @@ void Document::run_unloading_cleanup_steps()
...
 
-    // 3. Run any unloading document cleanup steps for document that are defined by this specification and other applicable specifications.
+    // 4. Let ports be the list of MessagePorts whose relevant global object's associated Document is document.
+    // 5. For each port in ports, disentangle port.
+    HTML::MessagePort::for_each_message_port([&](HTML::MessagePort& port) {
+        auto& global = HTML::relevant_global_object(port);
+        if (!is<HTML::Window>(global))
+            return;
+
+        auto& window = static_cast<HTML::Window&>(global);
```
Comment (ADKaster) [inline Userland/Libraries/LibWeb/DOM/Document.cpp:3327]: makes me wonder what happened to that idea for
```
if (auto* window = check_cast<HTML::Window>(global)) {
  // use window
}
```

**Pull request merged**